src/libasr/runtime/lfortran_intrinsics.c

```diff
@@ -205,14 +217,15 @@ void handle_float(char* format, double val, char** result) {
     char int_str[64];
     sprintf(int_str, "%ld", integer_part);
     char dec_str[64];
-    sprintf(dec_str, "%f", decimal_part);
+    // TODO: This will work for up to `F65.60` but will fail for:
+    // print "(F67.62)", 1.23456789101112e-62_8
+    sprintf(dec_str, "%.*lf", (60-integer_length), decimal_part);
     memmove(dec_str,dec_str+2,strlen(dec_str));
 
     char* dot_pos = strchr(format, '.');
+    decimal_digits = atoi(++dot_pos);
     width = atoi(format + 1);
     if (dot_pos != NULL) {
-        dot_pos++;
```

Comment:
@gptsarthak can you review the changes here related to the formatting.

Reply:
I checked with a diffchecker and i dont see any difference between this and lfortran main, so it looks good to me.

Reply:
Good, thanks for the review!